### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by praveek**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by prudrabhat — APPROVED** [review]

File: Documentation/GettingStarted.md [hunk 1]
```diff
@@ -16,16 +16,16 @@ If you cannot access the Mobile Install Instructions dialog box in the Data Coll
...
-implementation 'com.adobe.marketing.mobile:signal:2.x.x'
-implementation 'com.adobe.marketing.mobile:lifecycle:2.x.x'
+implementation 'com.adobe.marketing.mobile:userprofile:2.+'
+implementation 'com.adobe.marketing.mobile:core:2.+'
+implementation 'com.adobe.marketing.mobile:identity:2.+'
+implementation 'com.adobe.marketing.mobile:signal:2.+'
+implementation 'com.adobe.marketing.mobile:lifecycle:2.+'
 ```
 
+> Using dynamic dependency versions is not recommended for production apps. Refer to this [page](./GradleDependencies.md) for managing gradle dependencies
```
Comment (emdobrin) [inline Documentation/GettingStarted.md:27]: - we can mark this as warning
```
> **Warning**
> Using dynamic ..
```
 - nit: close the sentence with period.

File: Documentation/GradleDependencies.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Managing Gradle Dependencies
+
+Our getting started guide and the Android installation instructions generated for Mobile Properties in [Data Collection UI](https://developer-stage.adobe.com/client-sdks/documentation/getting-started/get-the-sdk/) uses dynamic versioning for Gradle dependencies. This is done to simplyfy getting started with the latest versions of our SDKs
```
Comment (emdobrin) [inline Documentation/GradleDependencies.md:3]: Our x and y use dynamic ..

File: Documentation/GradleDependencies.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+Use of dynamic dependency version is not recommended as it makes builds non-deterministic. This causes builds to break without any obvious change, and worse, can be caused by a transitive dependency that the build author has no control over.
+
+To have predictable builds, follow one of these approaches when managing gradle dependencies. 
+
+## Resolve dependencies using IDE
+
+To ensure consistent builds, it is best to explicitly specify the versions of dependencies and manually update them as necessary. This allows for full control over version updates and the ability to thoroughly review the dependency changelog and assess the changes before implementation.
+
+IDEs such as Android Studio or IntelliJ will highlight dynamic or outdated dependencies as warnings. Follow the suggestions provided to update them to appropriate version.
```
Comment (emdobrin) [inline Documentation/GradleDependencies.md:13]: nit: use present tense: IDEs such as .. highlight dynamic

File: Documentation/GradleDependencies.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+![Dynamic dependency warning](./assets/dependencies_hint1.png)
+![Outdated dependency warning](./assets/dependencies_hint2.png)
+
+## Locking dependency versions
+
+Gradle dependency locking is a feature that allows developers to lock the versions of dependencies used in a project, ensuring that the same versions are used consistently across different builds and environments. This helps to ensure that the project remains stable and predictable, and prevents version conflicts or unexpected behavior caused by using different versions of dependencies.
```
Comment (emdobrin) [inline Documentation/GradleDependencies.md:20]: This helps to ensure that the project remains stable and predictable -> This increases build's stability and predictability, ..

File: README.md [hunk 5]
```diff
@@ -17,46 +17,45 @@ The `MobileCore` and `Identity` extensions represent the foundation of the Adobe
...
- | [AEP Assurance SDK for Android](https://github.com/adobe/aepsdk-assurance-android) |  Provides support to inspect and validate AEP SDK by integrating with Adobe Experience Platform Assurance. |
+Additional documentation about API usage and SDK architecture can be found under the [Documentation](./Documentation) directory.
 
-## Documentation
+## Related Projects
 
-Additional documentation for usage and SDK architecture can be found under the [Documentation](Documentation) directory.
+| Project | Maven | Github |
+|---|---|---|
+| [Profile](https://developer.adobe.com/client-sdks/documentation/profile/) | [![Maven Central](https://img.shields.io/maven-central/v/com.adobe.marketing.mobile/userprofile.svg?logo=android&logoColor=white&label=userprofile&style=flat-square)](https://mvnrepository.com/artifact/com.adobe.marketing.mobile/userprofile) | [Link](https://github.com/adobe/aepsdk-userprofile-android) |
```
Comment (emdobrin) [inline README.md:38]: Can we link just the github projects in this table and remove the docs one? Also some projects provide a short description for each extension, e.g.
| [Analytics extension](https://github.com/adobe/aepsdk-analytics-android)             | The Analytics extension allows for sending mobile interaction data from your app to Adobe Analytics. |
| [Assurance extension](https://github.com/adobe/aepsdk-assurance-android)             | The Assurance extension enables validation workflows for your SDK implementation.                    |

**Pull request merged**